openwisp_monitoring/monitoring/static/monitoring/js/chart.js

```diff
@@ -16,6 +16,128 @@
         }
         return newArr;
     }
+
+    function adaptiveFilterPoints(charts, yRawVal) {
+
+        var convertToAdaptiveBytes = function(charts, type, i, multiplier, unit) {
+            charts[type].y[i] = Math.round((charts[type].y[i] * multiplier) * 100) / 100;
+            charts[type].hovertemplate[i] = charts[type].y[i] + ' ' + unit;
+        };
+
+        for (var i=0; i<charts[0].y.length; i++) {
+            if (yRawVal[i] == null) {
+                charts[0].hovertemplate[i] = 'N/A' + '<extra></extra>';
+            }
+            else if (charts[0].y[i] == 0) {
+                convertToAdaptiveBytes(charts, 0, i, 1, 'B');
+            }
+            else if (charts[0].y[i] < 0.001) {
+                convertToAdaptiveBytes(charts, 0, i, 1000000, 'KB');
+            }
+            else if (charts[0].y[i] < 1) {
+                convertToAdaptiveBytes(charts, 0, i, 1000, 'MB');
+            }
+            else if (charts[0].y[i] < 1000) {
+                convertToAdaptiveBytes(charts, 0, i, 1, 'GB');
+            }
+            else {
+                convertToAdaptiveBytes(charts, 0, i, 0.001, 'TB');
+            }
+        }
+
+        for (i=0; i<charts[1].y.length; i++) {
+            if (yRawVal[i] == null) {
+                charts[1].hovertemplate[i] = 'N/A' + '<extra></extra>';
+            }
+            else if (charts[1].y[i] == 0) {
+                convertToAdaptiveBytes(charts, 1, i, 1, 'B');
+            }
+            else if (charts[1].y[i] < 0.001) {
+                convertToAdaptiveBytes(charts, 1, i, 1000000, 'KB');
+            }
+            else if (charts[1].y[i] < 1) {
+                convertToAdaptiveBytes(charts, 1, i, 1000, 'MB');
+            }
+            else if (charts[1].y[i] < 1000) {
+                convertToAdaptiveBytes(charts, 1, i, 1, 'GB');
+            }
+            else {
+                convertToAdaptiveBytes(charts, 1, i, 0.001, 'TB');
+            }
+        }
+
+        for (i=0; i<charts[2].y.length; i++) {
+            if (yRawVal[i] == null) {
+                charts[2].hovertemplate[i] = 'N/A' + '<extra></extra>';
+            }
+            else if (charts[2].y[i] == 0) {
+                convertToAdaptiveBytes(charts, 2, i, 1, 'B');
+            }
+            else if (charts[2].y[i] < 0.001) {
+                convertToAdaptiveBytes(charts, 2, i, 1000000, 'KB');
+            }
+            else if (charts[2].y[i] < 1) {
+                convertToAdaptiveBytes(charts, 2, i, 1000, 'MB');
+            }
+            else if (charts[2].y[i] < 1000) {
+                convertToAdaptiveBytes(charts, 2, i, 1, 'GB');
+            }
+            else {
+                convertToAdaptiveBytes(charts, 2, i, 0.001, 'TB');
+            }
+        }
+    }
+
+    function adaptiveFilterLayout(charts, layout) {
```

Comment:
@yashikajotwani12 it is not clear to me how this function is related to the `filter` as indicated by the name. Can you please explain it?

Reply:
Because the Y-axis layout has a specific unit that has been set by calculating the values of data points, basically filtering through the average of the set of values.